packages/jest/src/executors/jest/jest.impl.ts

```diff
@@ -2,23 +2,138 @@ import 'dotenv/config';
+  context: ExecutorContext,
+  dependencies: DependentBuildableProjectNode[]
+): string => {
+  const jestConfig = readFileSync(
+    path.resolve(context.cwd, jestConfigPath)
+  ).toString();
+
+  const overrides = createJestOverrides(root, context, dependencies);
+
+  return `
+    const path = require('path');
+    const { pathsToModuleNameMapper } = require('ts-jest/utils')
+    ${jestConfig}
+    ${overrides}
+  `;
+};
+
+/**
+ * Attempt to find the compiled index.js to use as the entry point for this library
+ *
+ * Built in library schematics are as follows for the package.json index file:
+ *
+ * Nest: 'main'
+ * Angular: 'esm2020', 'es2020', 'module' (No order, as of Angular 13 all three point to the same artifacts)
+ * Node: 'main'
+ * React: 'main', 'module' (Cannot use `module` as Jest does not support it yet)
+ * TS (@nrwl/js): 'main'
+ * Web: 'main'
+ *
+ * @param distPackageJson
+ */
+export function findIndexFile(distPackageJson: any): string | undefined {
+  if (distPackageJson['main']) {
+    return distPackageJson['main'];
+  }
+
+  return distPackageJson['esm2020'];
+}
+
+export function createJestOverrides(
+  root: string,
+  context: ExecutorContext,
+  dependencies: DependentBuildableProjectNode[]
+): string {
+  const offset = offsetFromRoot(root);
+
+  const tsAliasPrefix = `@${context.workspace.npmScope}/`;
+
+  const paths = dependencies
+    .filter((dep) => !dep.node.name.startsWith('npm:'))
+    .reduce((prev, dep) => {
+      const node = dep.node as ProjectGraphProjectNode;
+
+      const rootPath = node.data.root.split('/');
+      rootPath.shift();
+
+      const distPackageJson = readJsonFile(
+        path.join(context.cwd, 'dist', node.data.root, 'package.json')
+      );
+
+      const indexFile = findIndexFile(distPackageJson);
+
+      if (!indexFile) {
+        return prev;
+      }
+
+      return [
+        ...prev,
+        `"${tsAliasPrefix}${rootPath.join('/')}": [
+          path.join(__dirname, '../${offset}dist/', '${
+          node.data.root
+        }', '${indexFile}')
+        ]`,
+      ];
+    }, []);
+
+  return `
+  module.exports = {
+    ...module.exports,
+    preset: '${offset}jest.preset.js',
+    rootDir: path.join(__dirname, '../${offset}${root}'),
+    coverageDirectory: '../${offset}coverage/${root}',
+    moduleNameMapper: {
+      ...(module.exports.moduleNameMapper || {}),
+      ...pathsToModuleNameMapper({
+        ${paths.join(',')}
+      })
+    }
+  }`;
+}
+
 export async function jestExecutor(
   options: JestExecutorOptions,
   context: ExecutorContext
 ): Promise<{ success: boolean }> {
   const config = await jestConfigParser(options, context);
 
-  const { results } = await runCLI(config, [options.jestConfig]);
+  /*
+   * If config.config exists that means we are running tests using artifacts
+   * `testFromSource: false` otherwise default to the projects jest config
+   */
+  const { results } = await runCLI(config, [
+    config.config ?? options.jestConfig,
```

Comment:
this is the one thing I am a little unsure on. I have it working for a single jest run but I didn't implement it for batch testing. Does that need to be done? Does does batch testing even work right now with nx, I have never seen anything referring to it.